File: Backend/Event.php
<?php

require_once("ConnectionData.php");
require_once("class.RequestHandler.php");

class Event{
   public $id;
   public $title;
   public $maxNumberOfPeople;
   public $numberOfPeople;
   public $password;
   public $description;
   public $latitude;
   public $longitude;
   public $phoneNumber;
   public $isPrivate;
   public $contentID;
   public $customImage;
   public $dateTime;
   public $ownerId;

   function __construct($id ,$title, $maxNumberOfPeople, $numberOfPeople, $password, $description, $latitude,
   $longitude,$phoneNumber, $isPrivate, $contentID, $customImage, $dateTime, $ownerId)
   {
       $this->id = $id;
       $this->title = $title;
       $this->maxNumberOfPeople = $maxNumberOfPeople;
       $this->numberOfPeople = $numberOfPeople;
       $this->password = $password;
       $this->description = $description;
       $this->latitude = $latitude;
       $this->longitude = $longitude;
       $this->phoneNumber = $phoneNumber;
       $this->isPrivate = $isPrivate;
       $this->contentID = $contentID;
       $this->customImage = $customImage;
       $this->dateTime = $dateTime;
       $this->ownerId = $ownerId;
   }

   function insertNewEvent(){
        $connectionData = new ConnectionData();
        $connectionString = $connectionData->getConnectionString();
      
        
        if (mysqli_connect_errno())
        {
        echo "Failed to connect to MySQL: " . mysqli_connect_error();
        }

        $query = "INSERT INTO `Event` (`id`, `title`, `maxNumberOfPeople`, `numberOfPeople`, `password`, `description`, `latitude`, `longitude`, `phoneNumber`, 
        `isPrivate`, `contentID`, `customImage`, `dateTime`, `ownerId`) 
        VALUES (NULL, '$this->title', '$this->maxNumberOfPeople', NULL, '$this->password', '$this->description', '$this->latitude', '$this->longitude', NULL, '$this->isPrivate',
         '$this->contentID', NULL, '$this->dateTime', '$this->ownerId')";

        if(!mysqli_query($connectionString,$query)) 
        {
            echo mysqli_error($connectionString);
            die('Error : Query Not Executed. Please Fix the Issue! '); 
        
        }else
        {
            $newRecordId = mysql_insert_id();
            $sql  = "SELECT Event.* FROM `Event` WHERE id = '$newRecordId'";
            $requestHandler = new RequestHandler();
            $data = $requestHandler->getMultipleDataFromDatabase($sql);
            echo json_encode($data, JSON_NUMERIC_CHECK);
        }
        mysqli_close($connectionString);  
    }

    function updateEvent(){
        $connectionData = new ConnectionData();
        $connectionString = $connectionData->getConnectionString();
      
        
        if (mysqli_connect_errno())
        {
        echo "Failed to connect to MySQL: " . mysqli_connect_error();
        }

        $query = "UPDATE `Event` SET `title` = '$this->title', `maxNumberOfPeople` = '$this->maxNumberOfPeople', `password` = '$this->password', `description` = '$this->description', `latitude` = '$this->latitude', `longitude` = '$this->longitude', `isPrivate` = '$this->isPrivate', `contentID` = '$this->contentID', `dateTime` = '$this->dateTime' WHERE `Event`.`id` = '$this->id'";

        if(!mysqli_query($connectionString,$query)) 
        {
            echo mysqli_error($connectionString);
            die('Error : Query Not Executed. Please Fix the Issue! '); 
        
        }else
        {
            $updatedRecordId = $this->id;
            $sql  = "SELECT Event.* FROM `Event` WHERE id = '$updatedRecordId'";
            $requestHandler = new RequestHandler();
            $data = $requestHandler->getMultipleDataFromDatabase($sql);
            echo json_encode($data, JSON_NUMERIC_CHECK);
        }
        mysqli_close($connectionString);  
        
    }

}
?>

File: Client/AIR1912/Common/Models/Event.swift
//
//  Event.swift
//  AIR1912
//
//  Created by Leo Leljak on 13/12/2019.
//  Copyright © 2019 Leo Leljak. All rights reserved.
//

import Foundation



struct Event: Decodable, Encodable {
    
       var id: Int
       var title: String
       var maxNumberOfPeople: Int
       var numberOfPeople: Int?
       var password: String?
       var description: String
       var latitude: Double
       var longitude: Double
       var phoneNumber: String?
       var isPrivate: Int
       var contentID:Int
       var customImage: String?
       var dateTime: String
       var ownerId: Int
       
    
    
     init(title: String, maxNumberOfPeople: Int, numberOfPeople: Int?, password: String?, description: String, latitude: Double, longitude: Double, phoneNumber: String?, isPrivate: Int, contentID: Int, customImage: String?, dateTime: String, ownerId: Int) {
       
        self.id = 0
        self.title = title
        self.maxNumberOfPeople = maxNumberOfPeople
        self.numberOfPeople = numberOfPeople
        self.password = password
        self.description = description
        self.latitude = latitude
        self.longitude = longitude
        self.phoneNumber = phoneNumber
        self.isPrivate = isPrivate
        self.contentID = contentID
        self.customImage = customImage
        self.dateTime = dateTime
        self.ownerId = ownerId
    }
    
    init(title: String, maxNumberOfPeople: Int, description: String, latitude: Double, longitude: Double, isPrivate: Int, contentID: Int, dateTime: String, ownerId: Int) {
       
        self.id = 0
        self.title = title
        self.maxNumberOfPeople = maxNumberOfPeople
        self.numberOfPeople = 0
        self.password = nil
        self.description = description
        self.latitude = latitude
        self.longitude = longitude
        self.phoneNumber = nil
        self.isPrivate = isPrivate
        self.contentID = contentID
        self.customImage = nil
        self.dateTime = dateTime
        self.ownerId = ownerId
    }
   
    
    
    
}
